src/lib/operators/insert.cpp

```diff
@@ -209,12 +215,28 @@ void Insert::_on_rollback_records() {
     const auto target_chunk = _target_table->get_chunk(target_chunk_range.chunk_id);
     auto mvcc_data = target_chunk->get_scoped_mvcc_data_lock();
 
+    /**
+     * !!! Crucial comment, PLEASE READ AND _UNDERSTAND_ before altering any of the following code !!!
+     *
+     * Set end_cids to 0 (effectively making the rows invisible for everyone) BEFORE setting the begin_cids to 0.
+     *
+     * Otherwise, another thread might observe a row with `begin_cid == 0` and `end_cid == MAX_COMMIT_ID` which would
+     * make this other thread believe it can see a row that was actually rolled back and should never become visible.
```

Comment:
Mmh, this is not that easy to understand. Maybe:
Otherwise, another transaction/thread might observe a row with `begin_cid == 0`, `end_cid == MAX_COMMIT_ID` and a foreign tid which is what a visible row that is being deleted by a different transaction looks like. Thus, the other transaction would consider the row that is in the process of being rolled back and should have never been visible as visible.

Reply:
mmh, okay, changed.